engine/gamesys/src/gamesys/components/comp_collection_proxy.cpp

```diff
@@ -130,10 +130,13 @@ namespace dmGameSystem
     {
         CollectionProxyWorld* proxy_world = new CollectionProxyWorld();
         CollectionProxyContext* context = (CollectionProxyContext*)params.m_Context;
-        const uint32_t component_count = context->m_MaxCollectionProxyCount;
+        uint32_t component_count = MIN(params.m_MaxComponentIntances, context->m_MaxCollectionProxyCount);
         proxy_world->m_Components.SetCapacity(component_count);
         proxy_world->m_Components.SetSize(component_count);
-        memset(&proxy_world->m_Components[0], 0, sizeof(CollectionProxyComponent) * component_count);
+        if(component_count > 0)
```

Comment:
if we have 0 components we can't get &proxy_world->m_Components[0]

Reply:
We have "array.Begin()" to get the pointer of the internal array

Reply:
Is it ok if the component_count == 0?
Shouldn't that be a build error?
I.e. we want to alert the user that they're including empty content?

Reply:
yes, it's ok that component_count==0. User can set max_count to 0 to avoid allocation of memory for components he doesn't use. 

Also component_count can be zero because the build counter found that there is no such components in collection.

At the moment I create component worlds with zero components to avoid protective code for every function that may use it (except physics component).
In the future we can prevent creation of component worlds if there are no components in collection.
or even better is to perevent creation of the component in AllocCollection. It will help to have one protection for all the function (if we don't have message reciever that means collection doesn't have components) but in this case we need a separate register per collection to sort components by priority and call them one by one (another option is checking if ollection->m_ComponentWorlds[i] exist all the time)

Reply:
`array.Begin()` is perfectly fits here, thank you!